Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1153,6 +1212,13 @@ LayoutUnit RenderGrid::gridAreaBreadthForChildIncludingAlignmentOffsets(const Re
 {
     // We need the cached value when available because Content Distribution alignment properties
     // may have some influence in the final grid area breadth.
+    auto directionIsMasonryAxisDirection = [this](GridTrackSizingDirection direction) {
+        return (style().gridMasonryRows() && direction == ForRows) || (style().gridMasonryColumns() && direction == ForColumns);
```

Comment:
What if `style().gridMasonryColumns() && direction == ForColumns` is true but also `style().gridMasonryRows()`? Then only rows should be masonry.

Reply:
I added a couple of helper methods for this type of logic `isMasonryRows` is simple and just checks for `style.gridMasonryRows().` `isMasonryColumns` also does an additional check to make sure that rows is not set as well.